extensions/azure/data-plane-azure-storage/src/main/java/org/eclipse/dataspaceconnector/azure/dataplane/azurestorage/pipeline/AzureStorageDataSourceFactory.java

```diff
@@ -0,0 +1,102 @@
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  ~s://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.pipeline;
+
+import net.jodah.failsafe.RetryPolicy;
+import org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.adapter.BlobAdapterFactory;
+import org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.schema.AzureBlobStoreSchema;
+import org.eclipse.dataspaceconnector.dataplane.spi.pipeline.DataSource;
+import org.eclipse.dataspaceconnector.dataplane.spi.pipeline.DataSourceFactory;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+import org.eclipse.dataspaceconnector.spi.types.domain.DataAddress;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataFlowRequest;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashMap;
+import java.util.Objects;
+
+import static java.lang.String.format;
+import static org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.validator.AzureStorageValidator.validateAccountName;
+import static org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.validator.AzureStorageValidator.validateBlobName;
+import static org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.validator.AzureStorageValidator.validateContainerName;
+import static org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.validator.AzureStorageValidator.validateSharedKey;
+
+/**
+ * Instantiates {@link AzureStorageDataSource}s for requests whose source data type is {@link AzureBlobStoreSchema#TYPE}.
+ */
+public class AzureStorageDataSourceFactory implements DataSourceFactory {
+    private final BlobAdapterFactory blobAdapterFactory;
+    private final RetryPolicy<Object> retryPolicy;
+    private final Monitor monitor;
+
+    public AzureStorageDataSourceFactory(BlobAdapterFactory blobAdapterFactory, RetryPolicy<Object> retryPolicy, Monitor monitor) {
+        this.blobAdapterFactory = blobAdapterFactory;
+        this.retryPolicy = retryPolicy;
+        this.monitor = monitor;
+    }
+
+    @Override
+    public boolean canHandle(DataFlowRequest request) {
+        return AzureBlobStoreSchema.TYPE.equals(request.getSourceDataAddress().getType());
+    }
+
+    @Override
+    public @NotNull Result<Boolean> validate(DataFlowRequest request) {
+        var dataAddress = request.getSourceDataAddress();
+        var properties = new HashMap<>(dataAddress.getProperties());
+        try {
+            if (!Objects.equals(properties.remove(DataAddress.TYPE) , AzureBlobStoreSchema.TYPE)) {
+                // should never happen because of routing mechanism
+                throw new IllegalArgumentException("Unexpected type");
```

Comment:
Do we need this validation here ? As this already covered by canHandle flow via PipelineServiceImpl.validate method.

Reply:
Removed